### PR TITLE
Add segment analytics pump

### PR DESCRIPTION
**Pull request by JosephEarl**
Adds pump for Segment.com

### CODE REVIEW

File: pumps/segment.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+func (s *SegmentPump) WriteData(data []interface{}) error {
+	log.WithFields(logrus.Fields{
+		"prefix": segmentPrefix,
+	}).Info("Writing ", len(data), " records")
+
+	for i, v := range data {
+		analyticsRecord := AnalyticsRecord{}
+		err := msgpack.Unmarshal(v.([]byte), &analyticsRecord)
```
Comment (lonelycode) [inline pumps/segment.go]: Does this actually work?? You shouldn't need to unmarshal the data using msgpack, it's already unmarshalled when it hits this handler. There is a special case for uptime data, which issn't handled in other pumps. See the [influx handler](https://github.com/TykTechnologies/tyk-pump/blob/develop/pumps/influx.go#L92) for proof ;-)

Comment (JosephEarl) [inline pumps/segment.go]: I copied this over from some code I had against the 1.9 pump before it was separated out, so may not be needed anymore.

You are saying I can just do 
```
analyticsRecord := v.(AnalyticsRecord)
```
?

Comment (lonelycode) [inline pumps/segment.go]: Yup, no need to try and decode (that would generate an error I think).

**Comment by JosephEarl** [comment]
Removed the JSON decoding

**Comment by lonelycode** [comment]
Stupid question, but I gotta ask - but have you successfully used this with segment?

**Comment by JosephEarl** [comment]
I'm still doing some testing with Segment so I'd wait a bit before merging just to confirm it's OK.

**Comment by JosephEarl** [comment]
Updated and added a test (ignored currently - but you can enter your own Segment key and remove the Skip to try). Here's a screenshot of an event in the Segment debugger:

segmenttrackhit

**Comment by lonelycode** [comment]
Amazing! Will merge, should be available in the nightlies in about an hour (will publish URLs in the community forum soon)

**Comment by JosephEarl** [comment]
Great!

**Pull request merged**